sdv/data_processing/data_processor.py

```diff
@@ -380,7 +380,15 @@ def _create_config(self, data, columns_created_by_constraints):
                 self._anonymized_columns.append(column)
 
             elif sdtype in self._transformers_by_sdtype:
-                transformers[column] = deepcopy(self._transformers_by_sdtype[sdtype])
+                kwargs = {
+                    key: value for key, value in column_metadata.items()
+                    if key not in ['pii', 'sdtype']
+                }
+                if kwargs:
+                    transformer_class = self._transformers_by_sdtype[sdtype].__class__
+                    transformers[column] = transformer_class(**kwargs)
+                else:
+                    transformers[column] = deepcopy(self._transformers_by_sdtype[sdtype])
```

Comment:
minor: Should we move this logic into a private method so this method doesn't get too large?